nixos/modules/services/blockchain/nimbus/beacon-node.nix

```diff
@@ -0,0 +1,236 @@
+            default = 9100;
+            description = mdDoc "Metrics port for beacon node.";
+          };
+        };
+
+        rest = {
+          enable = lib.mkEnableOption (mdDoc "Nimbus beacon node REST API");
+          address = mkOption {
+            type = types.str;
+            default = "127.0.0.1";
+            description = mdDoc "Listening address of the REST API server.";
+          };
+
+          port = mkOption {
+            type = types.port;
+            default = 5052;
+            description = mdDoc "Port for the REST API server.";
+          };
+        };
+
+        extraArgs = mkOption {
+          type = types.listOf types.str;
+          default = [];
+          example = ["--num-threads=1" "--graffiti=1337_h4x0r"];
+          description = mdDoc "Additional arguments passed to node.";
+        };
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = [
+      { assertion = length cfg.web3Urls > 0;
+        message = "Nimbus beacon node requires at least one Web3 URL to work!"; }
+    ];
+
+    users.users = optionalAttrs (cfg.service.user == "nimbus") {
+      nimbus = {
+        group = cfg.service.group;
+        home = cfg.dataDir;
+        description = "Nimbus beacon node service user";
+        isSystemUser = true;
+      };
+    };
+
+    users.groups = optionalAttrs (cfg.service.user == "nimbus") {
+      nimbus = { };
+    };
+
+    systemd.services.nimbus-beacon-node = {
+      description = "Nimbus Beacon Node (Ethereum consensus client)";
+
+      serviceConfig = mkMerge [{
+        User = cfg.service.user;
+        Group = cfg.service.group;
+
+        ExecStart = concatStringsSep " \\\n  " ([
+          "${cfg.package}/bin/nimbus_beacon_node"
+          "--network=${cfg.network}"
+          "--data-dir=${cfg.dataDir}"
+          "--log-level=${toUpper cfg.log.level}"
+          "--log-format=${cfg.log.format}"
+          "--nat=${cfg.nat}"
+          "--tcp-port=${toString cfg.listenPort}"
+          "--udp-port=${toString cfg.discoverPort}"
+          "--subscribe-all-subnets=${boolToString cfg.subAllSubnets}"
+          "--doppelganger-detection=${boolToString cfg.doppelganger}"
+          "--rest=${boolToString cfg.rest.enable}"
+        ] ++ optionals cfg.rest.enable [
+          "--rest-address=${cfg.rest.address}"
+          "--rest-port=${toString cfg.rest.port}"
+        ] ++ [
+          "--metrics=${boolToString cfg.metrics.enable}"
+        ] ++ optionals cfg.metrics.enable [
+          "--metrics-address=${cfg.metrics.address}"
+          "--metrics-port=${toString cfg.metrics.port}"
+        ] ++ (forEach cfg.web3Urls (u: "--web3-url=${u}"))
+          ++ optionals (cfg.jwtSecret != null) ["--jwt-secret=${cfg.jwtSecret}"]
+          ++ optionals (cfg.suggestedFeeRecipient != null) ["--suggested-fee-recipient=${cfg.suggestedFeeRecipient}"]
+          ++ (forEach cfg.extraArgs escapeShellArg));
+
+        WorkingDirectory = cfg.dataDir;
+        TimeoutSec = 1200;
```

Comment:
That is really really long timeout. Why not remove a zero?

Reply:
This is simply the timeout I coped from the official service config:
https://github.com/status-im/nimbus-eth2/blob/7dd4718735749362c50ad9d3d61a4b75e474d82c/scripts/package_src/nimbus_beacon_node/image/lib/systemd/system/nimbus_beacon_node.service#L54

@arnetheduck any idea why it's set so high? I see you committed that.

Reply:
It might be just a typo. I'm happy to lower it.

Reply:
Very well. Thanks for answering.